rasa/shared/utils/io.py

```diff
@@ -321,25 +322,106 @@ def env_var_constructor(loader, node):
     yaml.SafeConstructor.add_constructor("!env_var", env_var_constructor)
 
 
+class YAMLParser(yaml.YAML):
+    """A custom parser for YAML.
+
+    It's used because `ruamel.yaml` stores all its information globally.
+    Which means it's impossible to create different parsers with different constructors
+    and resolvers without remembering the previous configuration and restoring it once
```

Comment:
Is this threadsafe? Unfortunately we spawn new threads for some API endpoints...

Reply:
I updated and simplified it — now it's thread-safe as well.